gensim/test/test_nmf.py

```diff
@@ -87,18 +88,26 @@ def testTransform(self):
         transformed = self.model[doc]
 
         vec = matutils.sparse2full(transformed, 2)  # convert to dense vector, for easier equality tests
-        expected = [0., 1.]
+        expected = [0.02991635, 0.97008365]
+
         # must contain the same values, up to re-ordering
-        self.assertTrue(np.allclose(sorted(vec), sorted(expected)))
+        if os.name == 'nt':  # result is a bit off on Windows machines, relax tolerance in that case
```

Comment:
Please link to the discussion and investigation results by @Maocx here. "Result is a bit off" is unnecessarily cryptic for future maintainers, we can do better than that.

Reply:
You could be more terse and clear here.

```python
if os.name == 'nt':
    tolerance = ...
else:
    tolerance = ...
# or even use the ternary operator
self.assertTrue(np.allclose(..., rtol=tolerance))

Reply:
We could also include the platform-specific results here, these should not change between runs if the randomState has been set :) I can get a better precision both on Windows and Linux by increasing the h_stop_condition parameter. So something more akin to, which runs on both my windows machine and the virtual Linux:
```python
import logging
import os

import numpy as np
from gensim import matutils
from gensim.models import nmf
from gensim.test.utils import common_corpus, common_dictionary

logging.basicConfig(format='%(asctime)s : %(levelname)s : %(message)s', level=logging.INFO)

for i in range(10):

    model = nmf.Nmf(
        common_corpus,
        id2word=common_dictionary,
        chunksize=1,
        num_topics=2,
        passes=100,
        random_state=42,
        h_stop_condition=1e-7
    )

    word = 5
    transformed = model.get_term_topics(word)

    vec = matutils.sparse2full(transformed, 2)
    if os.name == 'nt':
        expected = [[0.3492566, 0.65074337]]
    else:
        linux = [[0.30723447, 0.69276553]]
    if not np.allclose(vec, expected):
        print("@@@@@@@@@@@@@@")
        break
```

Reply:
Doesn't seem to happen any more, thankfully.